### PR TITLE
issue-1 disposing of unused locks to prevent memory leaks

### PR DESCRIPTION
**Pull request by mrjobi**

### CODE REVIEW

File: src/main/java/pl/mj/valuelock/implementation/ValueMutex.java [hunk 1]
```diff
@@ -17,13 +19,21 @@ public void lock() {
...
+
     public boolean isLocked() {
         return delegate.isLocked();
     }
 
     int getQueueLength() {
         return delegate.getQueueLength();
     }
+
+    private void disposeIfNeccessary() {
```
Comment (pawel-kaminski-krk) [inline src/main/java/pl/mj/valuelock/implementation/ValueMutex.java]: is this thread safe?

Comment (mrjobi) [reply src/main/java/pl/mj/valuelock/implementation/ValueMutex.java]: no, it's not. (now it is, in master branch)

Comment (pawel-kaminski-krk) [reply src/main/java/pl/mj/valuelock/implementation/ValueMutex.java]: shouldn't it be thread safe, If you try to dispose while other thread acquires same lock and another thread after disposal. you will end up with two locks for same value ... Am I wrong?

Comment (mrjobi) [reply src/main/java/pl/mj/valuelock/implementation/ValueMutex.java]: yes, I fixed it on the master branch already

**Pull request merged**